server/src/main/java/org/elasticsearch/index/analysis/AnalyzerComponents.java

```diff
@@ -44,6 +44,12 @@ static AnalyzerComponents createComponents(
         final Map<String, CharFilterFactory> charFilters,
         final Map<String, TokenFilterFactory> tokenFilters
     ) {
+        for (String key : analyzerSettings.keySet()) {
```

Comment:
one additional suggestion: this is technically a breaking change, as existing indices will not be loaded if they have have an unknown param in the definition of a custom analyzer. We could mitigate this by making the new behaviour depend on the index created version, so that we'd throw error only for newly created indices. We'd need to have the index created version propagated from AnalysisRegistry though.

Reply:
Sorry. I am late. But i find the `method of createComponents` is stateless. How can we propagate the index created version to it?

Reply:
OK. I got it.